test/parallel/test-http-hostname-typechecking.js

```diff
@@ -21,7 +21,7 @@ vals.forEach((v) => {
 // These values are OK and should not throw synchronously
 ['', undefined, null].forEach((v) => {
   assert.doesNotThrow(() => {
-    http.request({hostname: v}).on('error', common.mustCall()).end();
-    http.request({host: v}).on('error', common.mustCall()).end();
+    http.request({hostname: v}).on('error', () => {}).end();
```

Comment:
Maybe add a comment that we don't care about the connection just about hostname type checking?
Maybe even go as far as:
```
const dontCare = () => {};
http.request({hostname: v}).on('error', dontCare).end();
http.request({host: v}).on('error', dontCare).end();
```
Just so no one revert-reverts this